### ISSUE TITLE
Running on a graph with missing labels

### ISSUE THREAD

**Issue by jdcc** [klicperajo/ppnp#9]
Judging from [this code](https://github.com/klicperajo/ppnp/blob/master/ppnp/data/sparsegraph.py#L68) and the fact that the labels need to be an array of ints (so no NaNs), it looks like this repo assumes a graph with a ground truth label for every node (and then hides some of those labels from training passes). I'm interested in running this on a graph for which I don't have ground truth for every node. So I have a couple questions:

1. Is the above understanding correct?
2. If so, before I really get into the code, any pointers on what I might need to change to make this work?

Thanks for all the work you've done making this open!

**Comment by gasteigerjo** [comment]
The code just expects a matrix that has the right shape. You could fill the missing labels with some placeholder, `-1` for example. Then you just need to make sure to pass the right indices for the training set etc., so you don't encounter these placeholders (`-1` would ensure that the model always fails loudly, since `nll_loss` expects a positive integer). The indices are generated [here](https://github.com/klicperajo/ppnp/blob/master/ppnp/pytorch/training.py#L34).

Happy coding!

**Comment by jdcc** [comment]
Ah, perfect! Thanks!
